src/report/actions.js

```diff
@@ -0,0 +1,29 @@
+import { utils } from '../common';
+
+const { AsyncActionType } = utils;
+
+export const FETCH_REPORTS = new AsyncActionType('GET', 'FETCH_REPORTS');
+
+export const fetchReports = programKey => ({
+  type: FETCH_REPORTS.BASE,
+  payload: {
+    programKey,
+  },
+});
+
+export const fetchReportsSuccess = (programKey, reportData) => ({
+  type: FETCH_REPORTS.SUCCESS,
+  payload: {
+    programKey,
+    reportData,
+  },
+});
+
+export const fetchReportsFailure = error => ({
+  type: FETCH_REPORTS.FAILURE,
+  payload: { error },
+});
+
+export const noReportData = () => ({
```

Comment:
It doesn't seem like this action is handled by a reducer; it doesn't change the Redux state. I wonder if you really need this action. Is there a reason you included it?

Reply:
you are right, Michael, I'll remove it.